src/WeekCalendar/Cell.tsx

```diff
@@ -0,0 +1,47 @@
+import { Event } from '../types';
+import CalendarEvent from './CalendarEvent';
+import { useRef } from 'react';
+import styles from './weekCalendar.module.css';
+import { MILLISECONDS } from '../constants';
+
+interface CellProps {
+  cellDate: Date;
+  onCellClick: (date: Date) => void;
+  events: Event[];
+}
+
+const Cell = ({ cellDate, onCellClick, events }: CellProps) => {
+  const cellRef = useRef<HTMLDivElement>(null);
+  const cellWidth = (cellRef.current?.offsetWidth ?? 0) - 10;
+  const cellHeight = cellRef.current?.offsetHeight;
+
+  return (
+    <div
+      className={styles.cell}
+      onClick={() => onCellClick(cellDate)}
+      ref={cellRef}
+    >
+      {events?.map((event, overlappingEventsCount) => {
+        const startDateTime = new Date(event.startDateTime);
```

Comment:
since you perform this quite a few times it might be worth to consider this:
- once data is fetched from db, all string dates are converted to Date objects
- when saving event - date is converted from Date object to string

by doing this, every time you need event.startDateTime it will be Date object

Reply:
Tried doing that, but I think it's better this way than doing extra code conversions, also i chose not to go with it because for some reason if the property of object is Date, u still need to initialize it with new Date(event.startDateTime), if you don't, it doesn't allow to access getTime(), so I could redo if it's a must, but I prefer staying it  this way as it looks less code to me.